source/materials/OpticalMaterialProperties.cc

```diff
@@ -74,10 +74,8 @@ G4MaterialPropertiesTable* OpticalMaterialProperties::EpoxyFixedRefr(G4double n)
 
   XenonLiquidProperties LXe_prop;
 
-  const G4int ri_entries = 18;
-  G4double ri_energy[ri_entries] = {optPhotMinE_, 2*eV, 3*eV, 4*eV, 5*eV, 6*eV, 6.2*eV,
-                                    6.4*eV, 6.6*eV, 6.8*eV, 7*eV, 7.2*eV, 7.4*eV,
-                                    7.6*eV, 7.8*eV, 8*eV, 8.2*eV, optPhotMaxE_};
+  const G4int ri_entries = 2;
+  G4double ri_energy[ri_entries] = {optPhotMinE_, optPhotMaxE_};
```

Comment:
Please eradicate all use of C-arrays throughout the code base. In almost every case you'll want to use `std::vector` instead. This has been the case since before 1998 and even Geant4 has now understood this (in some places) in 10.7!

Reply:
Yes, this is something to be done. What about putting it off to a following PR, when we update nexus to GEANT4.10.07? This way we will be able to use directly the new signature of the `AddProperty()` function.

Reply:
Are there any plans to move to 10.7?

I would say that it is reasonable to leave any C-array using lines that are not directly touched by this PR. But for any lines that *are* modified, I recommend that the C-arrays be changed to `std::vector` right now. This will require changing the declaration, and the call to `AddProperty`, so 2 or 3 other lines. If it turns out to require changing dozens of lines, then leave it, but if it's only a few, please do it now.

We can be pretty sure that 

+ A concerted effort to replace C-arrays with `std::vector` in a subsequent PR will never happen, because we'll always have something more important to do. (Repaying technical debt isn't very sexy.)
+ People will end up copy-pasting this. So let's give them a chance to copy-paste the correct solution, rather than the wrong one. Let's plant the seed of the correct solution. You know what the correct solution is, because I told you; everyone else on the project will think that the C-array is the correct solution, because nowhere is there any indication to the contrary: the same mistake is repeated *everywhere* in the project. Here is our chance to record, in at least one place in the project, what the correct approach is.